packages/mui-base/src/ModalUnstyled/ModalManager.test.js

```diff
@@ -131,6 +131,25 @@ describe('ModalManager', () => {
       expect(fixedNode.style.paddingRight).to.equal('14px');
     });
 
+    it('should handle the scroll on custom scroll target', () => {
+      const html = document.querySelector('html');
+      html.style.paddingRight = '32px';
+
+      const modal = {};
+      modalManager.add(modal, container1);
+      modalManager.mount(modal, {
+        scrollLockContainer: html,
+      });
+      expect(container1.style.overflow).to.equal('');
+      expect(container1.style.paddingRight).to.equal('20px');
+      expect(html.style.overflow).to.equal('hidden');
+      // TODO: this should not be 32px, but 32+scrollbarSize
+      expect(html.style.paddingRight).to.equal('32px');
+      modalManager.remove(modal);
+      expect(html.style.overflow).to.equal('');
+      expect(html.style.paddingRight).to.equal('32px');
+    });
```

Comment:
I wouldn't trust this test, I think that it's the whole Modal that should be tested, especially for this tricky logic. I iterated a LOT of time on the current one.